### PR TITLE
chore(db): queue application version transformer migration to database

### PR DESCRIPTION
**Pull request by miguel-crespo-fdc**

### CODE REVIEW

File: database/migrations/sqlite/1718182822233211_deployment_attempts.up.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE IF NOT EXISTS deployment_attempts
+(
+    eslVersion INTEGER, -- internal ID for ESL
+    created TIMESTAMP,
+    envName VARCHAR,
+    appName VARCHAR,
+    queuedVersion BIGINT NULL,
```
Comment (sven-urbanski-freiheit-com) [inline database/migrations/sqlite/1718182822233211_deployment_attempts.up.sql:7]: I think the other table calls it `releaseVersion`, so we should call it `queuedReleaseVersion` here.

File: pkg/db/db.go [hunk 2]
```diff
@@ -3081,3 +3118,230 @@ func (h *DBHandler) processAllTeamLocksRow(ctx context.Context, err error, rows
...
+		row, err = h.processSingleDeploymentAttemptsRow(rows)
+		if err != nil {
+			return nil, err
+		}
+	}
+	err = closeRows(rows)
+	if err != nil {
+		return nil, err
+	}
+	return row, nil // no rows, but also no error
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/db.go:3324]:
```suggestion
	return row, nil
```
misleading comment

File: pkg/db/db.go [hunk 3]
```diff
@@ -3081,3 +3118,230 @@ func (h *DBHandler) processAllTeamLocksRow(ctx context.Context, err error, rows
...
+		"SELECT eslVersion, created, envName, appName, queuedVersion FROM deployment_attempts ORDER BY eslVersion DESC LIMIT 1;")
+
+	span.SetTag("query", insertQuery)
+	rows, err := tx.QueryContext(
+		ctx,
+		insertQuery)
+	return h.processDeploymentAttemptsRow(ctx, rows, err)
+}
+
+func (h *DBHandler) DBSelectQueuedDeploymentHistory(ctx context.Context, tx *sql.Tx, environmentName, appName string, limit int) ([]QueuedDeployment, error) {
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/db.go:3150]: Please rename this to be consistent with other functions, e.g. all should contain "DeploymentAttempt"

File: services/cd-service/pkg/repository/repository.go [hunk 4]
```diff
@@ -1865,24 +1865,54 @@ func (s *State) DeleteDirIfEmpty(directoryName string) (SuccessReason, error) {
...
+
+	if err != nil || queuedDeployment == nil {
+		return nil, err
+	}
+
+	var v uint64
+	if queuedDeployment.Version != nil {
+		v = uint64(*queuedDeployment.Version)
+	} else {
+		return nil, nil
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/repository.go:1879]: either replace this `return` with  ` v = nil` 
or
replace the `then` branch with `return ...`
for consistency

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
.

File: pkg/db/db_test.go [hunk 5]
```diff
@@ -849,6 +849,230 @@ func TestDeleteApplicationLock(t *testing.T) {
...
+	}
+
+	dbHandler, err := Connect(*dbConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return dbHandler
+}
+
+// @@@@@
```
Comment (sven-urbanski-freiheit-com) [inline pkg/db/db_test.go:1075]: what is this comment?

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: pkg/db/db_test.go (hunk 5 above)
Comment (miguel-crespo-fdc) [reply pkg/db/db_test.go:1075]: remnants  of a challenging merge conflict. Deleting it.

**Pull request merged**